src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleFormatterConfigureOptions.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging.Configuration;
+using Microsoft.Extensions.Logging.Console;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Configures a ConsoleFormatterOptions object from an IConfiguration.
+    /// </summary>
+    /// <remarks>
+    /// Doesn't use ConfigurationBinder in order to allow ConfigurationBinder, and all its dependencies,
+    /// to be trimmed. This improves app size and startup.
+    /// </remarks>
+    [UnsupportedOSPlatform("browser")]
+    internal sealed class ConsoleFormatterConfigureOptions : IConfigureOptions<ConsoleFormatterOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public ConsoleFormatterConfigureOptions(ILoggerProviderConfiguration<ConsoleLoggerProvider> providerConfiguration)
+        {
+            _configuration = providerConfiguration.GetFormatterOptionsSection();
+        }
+
+        public void Configure(ConsoleFormatterOptions options) => Bind(_configuration, options);
+
+        public static void Bind(IConfiguration configuration, ConsoleFormatterOptions options)
+        {
+            if (configuration["IncludeScopes"] is string includeScopes)
+            {
+                options.IncludeScopes = bool.Parse(includeScopes);
```

Comment:
We have good tests in place for things like what happens if parsing fails?

Reply:
In configuration binding the TypeConverters are used and can throw which I believe we wrap the exception inside InvalidOperationException.

Reply:
> We have good tests in place for things like what happens if parsing fails?

No, but I can add some tests.

> which I believe we wrap the exception inside InvalidOperationException.

I don't think we should strive to maintain the same exception types when the configuration is mis-configured in this scenario. The configuration is bound lazily, and so the exception doesn't occur until someone actually tries getting an `ILogger` service. It is not predictable when the exception will be thrown, so it wouldn't be possible for someone to try to `catch` the InvalidOperationException. The important piece is that an actionable/understandable exception is thrown.

Reply:
The one piece where the existing exception is useful is that passes in the configuration `path` to the exception message. So you get an exception like:

```
System.InvalidOperationException: Failed to convert configuration value at 'FormatterOptions:SingleLine' to type 'System.Boolean'.
 ---> System.FormatException: not a bool is not a valid value for Boolean.
 ---> System.FormatException: String 'not a bool' was not recognized as a valid Boolean.
   at System.Boolean.Parse(ReadOnlySpan`1 value)
   at System.Boolean.Parse(String value)
```

`FormatterOptions:SingleLine` is the key piece of info we would be missing. Versus the current PR behavior:

```
Unhandled exception. System.FormatException: String 'not a bool' was not recognized as a valid Boolean.
   at System.Boolean.Parse(ReadOnlySpan`1 value)
   at System.Boolean.Parse(String value)
   at Microsoft.Extensions.Logging.SimpleConsoleFormatterConfigureOptions.Configure(SimpleConsoleFormatterOptions options) in C:\git\runtime\src\libraries\Microsoft.Extensions.Logging.Console\src\SimpleConsoleFormatterConfigureOptions.cs:line 40
```

Reply:
I've addressed this with the latest commit. PTAL.